### PR TITLE
feat(core): add `selectedOptionsOrder` in `useSelect`

### LINKED ISSUE #6061: [FEAT]: Have ability to sort `selectedOptions` first in `useSelect`
### Is your feature request related to a problem? Please describe.

Currently we use `useSelect` with `defaultValue` to add extra options in our select list. But when `options` have more records like 30-50 records so the `selectedOptions` is at the bottom. 

We want `selectedOptions` at first position of the list.

```ts
useSelect({
  defaultValue: 1, // or [1, 2]
});
```

### Describe alternatives you've considered

First option, we can switch the order of  `selectedOptions` first then `options`.
image

Second option, have a property to sort by `selectedOptions` or `options` first.

```ts
useSelect({
  defaultValue: 1, 
 selectedOptionsSort: 'asc' | 'desc'  // <--- something like that, we can have better name than this.
});
```

### Additional context

At the bove.

### Describe the thing to improve

N/A

### PR DESCRIPTION
**Pull request by Dominic-Preap**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://refine.dev/docs/guides-concepts/contributing/#commit-convention

## Bugs / Features

- [x] Related issue(s) linked
- [x] Tests for the changes have been added
- [x] Docs have been added / updated
- [x] Changesets have been added https://refine.dev/docs/guides-concepts/contributing/#creating-a-changeset

## What is the current behavior?

Cannot sort `selectedOptions` at the top of list when use `useSelect` with `defaultValue`.

## What is the new behavior?

feat: add `selectedOptionsOrder` in `useSelect`

Fixes #6061

### CODE REVIEW

**Review by alicanerdurmaz — APPROVED** [review]

File: documentation/docs/ui-integrations/mantine/hooks/use-select/index.md [hunk 1]
```diff
@@ -149,6 +149,20 @@ useSelect({
 
 > For more information, refer to the [`useMany` documentation &#8594](/docs/data/hooks/use-many)
 
+### selectedOptionsOrder
```
Comment (aliemir) [inline documentation/docs/ui-integrations/mantine/hooks/use-select/index.md:152]: Documentation of other `useSelect` hooks and extensions should also include this prop:

| Hook              | File |
| ------------- | ------------- |
| `@refinedev/antd#useSelect` | documentation/docs/ui-integrations/ant-design/hooks/use-select/index.md |
| `@refinedev/antd#useRadioGroup`  | documentation/docs/ui-integrations/ant-design/hooks/use-radio-group/index.md  |
| `@refinedev/antd#useCheckboxGroup` | documentation/docs/ui-integrations/ant-design/hooks/use-checkbox-group/index.md |
| `@refinedev/antd#useCheckboxGroup` | documentation/docs/ui-integrations/ant-design/hooks/use-checkbox-group/index.md |

**Review by aliemir — CHANGES_REQUESTED** [review]
Hey @Dominic-Preap, thank you for the contribution! 🚀 There was a small issue in the test case, I pushed a fix for it! Other than that, I left a comment about the documentation and some adjustments that might be needed for the extensions of the `useSelect` hook.

Let us know if you can work on these as well 🙏

**Comment by Dominic-Preap** [comment]
Okay I'll update it tonight.

File: documentation/docs/ui-integrations/mantine/hooks/use-select/index.md (hunk 1 above)
Comment (Dominic-Preap) [reply documentation/docs/ui-integrations/mantine/hooks/use-select/index.md:152]: I have updated the related components & the docs. Let me know if I'm missing something.

**Review by aliemir — APPROVED** [review]
Thank you for the update @Dominic-Preap! I've sent a small update and added `@refinedev/antd` and `@refinedev/mui` to the changeset 🚀

**Pull request merged**